codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddBuiltinPlugins.java

```diff
@@ -56,18 +58,27 @@
                 RuntimeClientPlugin.builder()
                         .withConventions("@aws-sdk/middleware-sdk-api-gateway", "^0.1.0-preview.5",
                                         "AcceptsHeader", HAS_MIDDLEWARE)
-                        .servicePredicate((m, s) -> s.getId().getName().equals("BackplaneControlService"))
+                        .servicePredicate((m, s) -> testServiceId(s, "API Gateway"))
                         .build(),
                 RuntimeClientPlugin.builder()
                         .withConventions("@aws-sdk/middleware-sdk-s3", "^0.1.0-preview.2",
                                         "ValidateBucketName", HAS_MIDDLEWARE)
-                        .servicePredicate((m, s) -> s.getId().getName().equals("AmazonS3"))
+                        .servicePredicate((m, s) -> testServiceId(s, "S3"))
                         .build(),
                 RuntimeClientPlugin.builder()
                         .withConventions("@aws-sdk/middleware-expect-continue", "^0.1.0-preview.5",
                                         "AddExpectContinue", HAS_MIDDLEWARE)
-                        .servicePredicate((m, s) -> s.getId().getName().equals("AmazonS3"))
+                        .servicePredicate((m, s) -> testServiceId(s, "S3"))
+                        .build(),
+                RuntimeClientPlugin.builder()
+                        .withConventions("@aws-sdk/middleware-sdk-glacier", "^0.1.0-preview.7",
```

Comment:
One more thing: Can you add these middleware to `AwsDependency` class, this way we can easier maintain the dependencies

Reply:
fixed in 8766e3ef7341a929b8c5c395b2e2e55a509e6682